File: backend/app/application/instrument_type/commands/create_instrument_type.py
"""
Command: create instrument type.
"""

from dataclasses import dataclass


@dataclass(frozen=True)
class CreateInstrumentTypeCommand:
    """Create instrument type command."""

    name: str
    manufacturer: str | None = None
    model: str | None = None
    measurement_type: str | None = None
    accuracy_class: str | None = None
    verification_interval_months: int | None = None
    description: str | None = None

File: frontend/src/features/instrument-type/api/createInstrumentType.ts
import { http } from "../../../shared/api/http";
import { mapInstrumentType } from "./instrumentTypeMapper";
import type { InstrumentTypeRead } from "../model/types";
import type { InstrumentTypeApiDto } from "./types";

export interface CreateInstrumentTypeInput {
  name: string;
  manufacturer?: string | null;
  model?: string | null;
  measurement_type?: string | null;
  accuracy_class?: string | null;
  verification_interval_months?: number | null;
  description?: string | null;
}

export async function createInstrumentType(
  data: CreateInstrumentTypeInput,
): Promise<InstrumentTypeRead> {
  const response = await http.post<InstrumentTypeApiDto>(
    "/instrument-types/",
    data,
  );

  return mapInstrumentType(response.data);
}
